pkg/configuration/configuration.go

```diff
@@ -118,6 +118,24 @@ type TemplateRepositoryManifest struct {
 
 	// Arguments are a declaration of arguments to the template generator
 	Arguments map[string]Argument
+
+	// ModuleBlocks contains all of the module to module blocks that
+	// this module provides to other modules
+	ModuleBlocks struct {
```

Comment:
Not sure if I want to go with static defining of module blocks, so may remove this. It's a future optimization, IMO.